x/tokenomics/keeper/token_logic_modules.go

```diff
@@ -555,47 +655,81 @@ func (k Keeper) sendRewardsToAccount(
 // If this is not the case, then the supplier essentially did "free work" and the
 // actual claim amount is less than what was claimed.
 // Ref: https://arxiv.org/pdf/2305.10672
+// TODO_TEST: Add more tests for edge cases that exercise all the code paths
```

Comment:
You don't need to add the tests, but can you please create the test headers + comments on what the edge cases are.

NOW is the best time to document them.